core/src/main/java/io/grpc/Status.java

```diff
@@ -314,6 +317,32 @@ public static Status fromCodeValue(int codeValue) {
     }
   }
 
+  @SuppressWarnings("fallthrough")
+  private static Status fromCodeValue(byte[] asciiCodeValue) {
```

Comment:
Something like. Move the other code to `fromCodeValue0`. Virtually all status codes should be `Status.OK`. The method should be small enough to be always inlined (less than 35 bytecodes or so). Just an idea. Could give it a try :)
```java
if (asciiCodeValue.length == 1 && asciiCodeValue[0] == 0) {
  return Status.OK;
}
return fromCodeValue0(asciiCodeValue);
```